benchmarks/benchmarks/bench_function_base.py

```diff
@@ -98,44 +98,122 @@ def time_select_larger(self):
 class Sort(Benchmark):
     params = [
         ['quick', 'merge', 'heap'],
-        ['float32', 'int32', 'uint32']
+        ['float32', 'int32', 'uint32'],
+        [
+            'ordered',
+            'reversed',
+            'uniform',
+            'sorted_block_10',
+            'sorted_block_100',
+            'sorted_block_1000',
+            'swapped_pair_1_percent',
+            'swapped_pair_10_percent',
+            'swapped_pair_50_percent',
+            'random_unsorted_area_50_percent',
+            'random_unsorted_area_10_percent',
+            'random_unsorted_area_1_percent',
+            'random_bubble_1_fold',
+            'random_bubble_5_fold',
+            'random_bubble_10_fold',
+        ],
     ]
-    param_names = ['kind', 'dtype']
+    param_names = ['kind', 'dtype', 'array_type']
 
-    def setup(self, kind, dtype):
-        self.e = np.arange(10000, dtype=dtype)
-        self.o = np.arange(10001, dtype=dtype)
-        np.random.seed(25)
-        np.random.shuffle(self.o)
-        # quicksort implementations can have issues with equal elements
-        self.equal = np.ones(10000, dtype=dtype)
-        self.many_equal = np.sort(np.arange(10000) % 10).astype(dtype)
+    ARRAY_SIZE = 10000
 
-        try:
-            np.sort(self.e, kind=kind)
-        except TypeError:
-            raise NotImplementedError()
+    def setup(self, kind, dtype, array_type):
+        np.random.seed(1234)
+        self.arr = getattr(self, 'array_' + array_type)(self.ARRAY_SIZE, dtype)
 
-    def time_sort(self, kind, dtype):
-        np.sort(self.e, kind=kind)
+    def time_sort_inplace(self, kind, dtype, array_type):
+        self.arr.sort(kind=kind)
 
-    def time_sort_random(self, kind, dtype):
-        np.sort(self.o, kind=kind)
+    def time_sort(self, kind, dtype, array_type):
```

Comment:
Don't know if it is worth benchmarking the function in addition to inplace.

Reply:
In radix sort, an auxiliary array is allocated, and so the relative "cost" of this auxiliary array is different in both cases. (it measures the impact of an extra `malloc`)

That said, I also kept this because both were there in the original benchmark.